extensions/ui/org.eclipse.smarthome.ui.basic/web-src/smarthome.js

```diff
@@ -1192,21 +1192,32 @@
 				item: _t.item,
 				value: _t.input.checked ? "ON" : "OFF"
 			}));
-			_t.suppressUpdate();
 		});
 
+		_t.hasValue = _t.parentNode.getAttribute("data-has-value") === "true";
+		_t.valueNode = _t.parentNode.parentNode.querySelector(o.formValue);
+
 		_t.setValuePrivate = function(value, itemState) {
 			var
 				val = itemState === "ON";
 
-			_t.input.checked = val;
-
-			if (val) {
+			if (_t.input.checked !== val) {
+				_t.input.checked = val;
+				if (val) {
 				_t.parentNode.MaterialSwitch.on();
```

Comment:
I have big difficulties with indentation. Eclipse, even after updating the settings is adding a mix of tabs and spaces. If I use notepad to edit the file, after saving, it looks like the beginning of the file is updated !
Is there a reason to use tabs instead of spaces in JS file while we are using spaces for Java files ?
What setting did you change in Eclipse ?
I spent almost more time trying to have a correct indentation than working on the fix !!!